src/v/raft/recovery_stm.cc

```diff
@@ -488,23 +488,32 @@ recovery_stm::dispatch_append_entries(append_entries_request&& r) {
 }
 
 bool recovery_stm::is_recovery_finished() {
-    if (_ptr->_bg.is_closed()) {
```

Comment:
is this not checked because one of the other checks implies the gate is closed?

Reply:
the abort source check is used instead, i can add a gate check for completeness, in general when gate is closed abort must has been called earlier.

Reply:
seems ok to me just wasn't sure